src/main.cpp

```diff
@@ -44,7 +45,10 @@ static void repl(int argc, char *argv[])
             exit(0);
         } else {
             try {
-                exec(compile(parse(tokenize(s)), nullptr), argc, argv);
+                auto ast = parse(tokenize(s));
```

Comment:
You're going to have to rebase this against the current code before it can be merged.